consumers/writers/influxdb/consumer_test.go

```diff
@@ -118,119 +118,141 @@ func queryDB(fluxQuery string) (int, error) {
 }
 
 func TestSaveSenml(t *testing.T) {
-	for i := 0; i < 2; i++ {
-		// Testing both async and sync
-		repo := writer.New(client, repoCfg, i == 0)
-
-		cases := []struct {
-			desc         string
-			msgsNum      int
-			expectedSize int
-		}{
-			{
-				desc:         "save a single message",
-				msgsNum:      1,
-				expectedSize: 1,
-			},
-			{
-				desc:         "save a batch of messages",
-				msgsNum:      streamsSize,
-				expectedSize: streamsSize,
-			},
-		}
+	syncRepo := writer.NewSync(client, repoCfg)
+	asyncRepo := writer.NewAsync(client, repoCfg)
+
+	cases := []struct {
+		desc         string
+		msgsNum      int
+		expectedSize int
+	}{
+		{
+			desc:         "save a single message",
+			msgsNum:      1,
+			expectedSize: 1,
+		},
+		{
+			desc:         "save a batch of messages",
+			msgsNum:      streamsSize,
+			expectedSize: streamsSize,
+		},
+	}
 
-		for _, tc := range cases {
-			err := resetBucket()
-			assert.Nil(t, err, fmt.Sprintf("Cleaning data from InfluxDB expected to succeed: %s.\n", err))
-			now := time.Now().UnixNano()
-			var msgs []senml.Message
-
-			chanID, err := idProvider.ID()
-			assert.Nil(t, err, fmt.Sprintf("got unexpected error: %s\n", err))
-			pubID, err := idProvider.ID()
-			assert.Nil(t, err, fmt.Sprintf("got unexpected error: %s\n", err))
-			for i := 0; i < tc.msgsNum; i++ {
-				msg := senml.Message{
-					Channel:    chanID,
-					Publisher:  pubID,
-					Protocol:   "http",
-					Name:       "test name",
-					Unit:       "km",
-					UpdateTime: 5456565466,
-				}
-				// Mix possible values as well as value sum.
-				count := i % valueFields
-				switch count {
-				case 0:
-					msg.Subtopic = subtopic
-					msg.Value = &v
-				case 1:
-					msg.BoolValue = &boolV
-				case 2:
-					msg.StringValue = &stringV
-				case 3:
-					msg.DataValue = &dataV
-				case 4:
-					msg.Sum = &sum
-				}
-
-				msg.Time = float64(now)/float64(1e9) - float64(i)
-				msgs = append(msgs, msg)
-			}
+	// Test async
```

Comment:
Separate Sync and async tests, since it's also separated on the interface level (`TestSyncConsumer` and `TestAsyncConsumer`).